### PR TITLE
fix(curriculum): add element to create-a-stack tests

### LINKED ISSUE #42322: Create a Stack Class: peek() method should be tested with more than one element in the stack
**Affected page**
https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/create-a-stack-class

**Forum post**
https://forum.freecodecamp.org/t/create-a-stack-class-a-bug-in-tests-required-implementation-of-peek-method/462776

**The problem**
When solving the challenge, the code below passes the tests:

```js
this.peek = function () {
  return collection[0];
};
```

**Expected behaviour**
The tests should fail for anything other than the correct solution, which is:
```js
this.peek = function () {
  return collection[collection.length - 1];
};
```

**Proposed solution**
In the test case, push one more element to the stack:

```js
assert(
  (function () {
    var test = new Stack();
    test.push('PSY44');
    test.push('CS50');
    return test.peek() === 'CS50';
  })()
);
```

**System (please complete the following information):**
 - Device: MacBook Pro
 - OS: macOS 11.3.1 
 - Browser: chrome
 - Version: 90

### PR DESCRIPTION
**Pull request by nayanadasgupta**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42322

Changes:

Changed the pop and peek tests by adding an additional item to the stack. This will ensure that the user's implementation must correctly return the item at the top of the stack.

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
Hey there,

Thank you, for this PR! Just wanted to mention, we recommend working off of a new branch (not `main`), otherwise `main` becomes spoilt, once this is merged, and makes creating concurrent PRs more difficult for yourself. Also, we recommend a specific style of PR title and commit message, just to help the reviewing process, and general bookkeeping.

This is outlined in better detail here: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request

There is nothing you need to do to this one, but keep it in mind for future contributions. Thank you

**Review by gikf — DISMISSED** [review]
Thanks for noticing `pop` method also could be improved in similar way. LGTM :+1: 

I'd consider adding another method call to both tests - to confirm `peek` didn't remove element, and that `pop` did remove.

**Comment by nayanadasgupta** [comment]
> Thanks for noticing `pop` method also could be improved in similar way. LGTM 👍
> 
> I'd consider adding another method call to both tests - to confirm `peek` didn't remove element, and that `pop` did remove.

Thanks! I've modified the tests to check that `peek` didn't remove the top element, and that `pop` did remove the top element.

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/create-a-stack-class.md [hunk 1]
```diff
@@ -89,8 +90,9 @@ The `pop` method should remove and return the top element of the stack
 assert(
   (function () {
     var test = new Stack();
+    test.push('CS61');
     test.push('CS50');
-    return test.pop() === 'CS50';
+    return test.pop() === 'CS50' && test.pop() == 'CS61';
```
Comment (gikf) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/create-a-stack-class.md:95]: Just a little nitpick - let's make second comparison also strict (`===`) for consistency.

Comment (nayanadasgupta) [reply curriculum/challenges/english/10-coding-interview-prep/data-structures/create-a-stack-class.md]: Whoops - that was a silly typo! Added another commit to resolve that.

**Review by gikf — APPROVED** [review]
Looks good! :+1:

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by ojeytonwilliams** [comment]
@nayanadasgupta  Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**